dotnet/src/Connectors/Connectors.Ollama/Settings/OllamaPromptExecutionSettings.cs

```diff
@@ -11,6 +11,7 @@ namespace Microsoft.SemanticKernel.Connectors.Ollama;
 /// <summary>
 /// Ollama Prompt Execution Settings.
 /// </summary>
+[JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
```

Comment:
The `FromExecutionSettings` method should preserve and restore function choice behavior similarly to how it's done for OpenAI [L341](https://github.com/microsoft/semantic-kernel/blob/918724793e05f70215795bf193ec33018989959c/dotnet/src/Connectors/Connectors.OpenAI/Settings/OpenAIPromptExecutionSettings.cs#L341). Otherwise, some internal state of function choice behavior may get lost during the serialization/deserialization process.

Reply:
Thanks, added missing UT to ensure it + additional behavior I was overlooking.